### PR TITLE
docs: disambiguate protocol version

### LINKED ISSUE #3030: Disambiguate protocol version, block version, app version
## Context

I find https://docs.cometbft.com/v0.34/spec/core/data_structures#version quite confusing. It claims that `version.block` is the protocol version but when I query `celestia-appd status`, I observe that protocol version is a wrapper around p2p, block, and app versions. Can the docs please disambiguate protocol version, block version, and app version? And ideally not use "protocol version" in the validation for `block.version`

```
$ celestia-appd status

{
  "NodeInfo": {
    "protocol_version": {
      "p2p": "8",
      "block": "11",
      "app": "1"
    },
```

Screenshot 2024-05-07 at 1 21 59 PM

### PR DESCRIPTION
**Pull request by melekes**
Closes #3030

---

#### PR checklist

- [ ] ~~Tests written/updated~~
- [ ] ~~Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)~~
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [x] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

**Comment by adizere** [comment]
Tagging @rootulp for a quick check here.

**Review by rootulp — APPROVED** [review]
LGTM! Thanks for the fast response

File: spec/abci/abci++_methods.md [hunk 1]
```diff
@@ -32,8 +32,8 @@ title: Methods
     | Name          | Type   | Description                            | Field Number |
     |---------------|--------|----------------------------------------|--------------|
     | version       | string | The CometBFT software semantic version | 1            |
-    | block_version | uint64 | The CometBFT Block Protocol version    | 2            |
-    | p2p_version   | uint64 | The CometBFT P2P Protocol version      | 3            |
+    | block_version | uint64 | The CometBFT Block version             | 2            |
+    | p2p_version   | uint64 | The CometBFT P2P version               | 3            |
```
Comment (cason) [inline spec/abci/abci++_methods.md:36]: Should we keep protocol here?

File: spec/abci/abci++_methods.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ title: Methods
     |---------------------|--------|-----------------------------------------------------|--------------|---------------|
     | data                | string | Some arbitrary information                          | 1            | N/A           |
     | version             | string | The application software semantic version           | 2            | N/A           |
```
Comment (cason) [inline spec/abci/abci++_methods.md:44]: What is the difference between `version` and `app_version` here?

File: spec/core/data_structures.md [hunk 3]
```diff
@@ -147,10 +147,10 @@ NOTE: that this is more specifically the consensus version and doesn't include i
 P2P Version. (TODO: we should write a comprehensive document about
 versioning that this can refer to)
 
-| Name  | type   | Description                                                                                                     | Validation                                                                                                         |
```
Comment (cason) [inline spec/core/data_structures.md:150]: Can we reword the previous paragraph?

Maybe we should only mention that the p2p version is not part of this `Version` field. The p2p version is only relevant at the p2p communication layer.

**Review by cason — APPROVED** [review]
+1

**Comment by melekes** [comment]
@mergify backport v0.34.x

**Comment by melekes** [comment]
@mergify backport v0.37.x

**Comment by melekes** [comment]
@mergify backport v0.38.x

**Pull request merged**